src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs

```diff
@@ -447,7 +448,7 @@ public void Serialize(XmlWriter xmlWriter, object o, XmlSerializerNamespaces nam
                     }
                     SerializePrimitive(xmlWriter, o, namespaces);
                 }
-                else if (ShouldUseReflectionBasedSerialization(_mapping))
+                else if (ShouldUseReflectionBasedSerialization(_mapping) || _isReflectionBasedSerializer)
```

Comment:
I am wondering if it is still necessary to check `ShouldUseReflectionBasedSerialization` here.

Reply:
No, we cannot remove it.  _isReflectionBasedSerializer is only being set in FromMappings in this case. When to create a serializer from a single soap mapping, it won't set this value, and it need use this method to check if it is a soap mapping. In addition, this method also check if the SerializationMode is reflectiononly or not.

Reply:
Would it be feasible to consolidate all the logic of using reflection based serialization in one place?

Reply:
It is better to seperate them. ShouldUseReflectionBasedSerialization method is a static method but _isReflectionBasedSerializer is not static.